actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -82,11 +81,19 @@ def format(formatter, filter = nil)
 
       private
 
-      def filter_routes(filter)
-        if filter
+      def filter_routes(options)
```

Comment:
I'm sorry I forgot to go back and update this with the ideas I had for the option parsing later. I'll do that now :smile:

I think it would be better to have the option parser do:

```ruby
opts.on("-c", "--controller [PATTERN]") do |pattern|
  options[:filter] = pattern.downcase.sub(/_?controller\z/, '')
end
```

Then later we can just assign `filter = options[:filter]` (or just use a keyword argument and save the line).

That's still as flexible as your proposal, but we only have to deal with one option. What do you think? :smile:

(If you think about it, what does `controller` really mean in the context of `filter_routes`? After all the other options are `filter` and `pattern` — `controller` sticks out like a sore thumb :grin:)